app/assets/javascripts/discourse/app/routes/preferences-profile.js

```diff
@@ -1,7 +1,23 @@
+import { action } from "@ember/object";
+import { service } from "@ember/service";
 import RestrictedUserRoute from "discourse/routes/restricted-user";
 
 export default class PreferencesProfile extends RestrictedUserRoute {
+  @service currentUser;
+
   setupController(controller, model) {
     controller.set("model", model);
   }
+
+  @action
+  willTransition(transition) {
+    super.willTransition(...arguments);
+
+    if (this.currentUser.needs_required_fields_check) {
```

Comment:
I think _maybe_ `currentUser` will be null here for anonymous users, try visiting someone's profile page as anon user to double check. Might be fine

Reply:
We also need to allow transitioning to an admin route, checking `transition?.to.name.startsWith("admin")` should be fine